lms/djangoapps/certificates/models.py

```diff
@@ -1264,3 +1264,43 @@ class Meta:
 
     def __str__(self):
         return str(self.arguments)
+
+@python_2_unicode_compatible
+class CertificateDateOverride(TimeStampedModel):
+    """
+    Model to manually override a given certificate date with the given date.
+
+    .. no_pii:
+    """
+    generated_certificate = models.OneToOneField(
+        GeneratedCertificate,
+        on_delete=models.CASCADE,
+        primary_key=True,
```

Comment:
Re: `primary_key=True` —

- "If True, this field is the primary key for the model", otherwise a primary key will be automatically generated in an `AutoField`, like we’re used to.
- This means that the primary key of the `CertificateDateOverride` object will always be the same as the primary key of its associated `GeneratedCertificate`.
- I can be talked out of this, but the Django docs examples for `OneToOne` relationships all have this as `True`, so my gut is to go with it. It underscores the relationship between the two models.
- "`primary_key=True` implies `null=False` and `unique=True`." Great! We don’t need to set those; this should be required and unique.
  - If you’re wondering about `blank`—which specifies if this field can be empty on forms (like in the admin)—we’d want that to be `False`; but I’m not bothering to set that because `blank=False` is the default. [See `blank` docs.](https://docs.djangoproject.com/en/3.1/ref/models/fields/#blank)

Docs: https://docs.djangoproject.com/en/3.1/ref/models/fields/#primary-key

Reply:
From discussion at standup: Hm. Some wariness around this. I don’t have a great case for it, only what I said above:
> the Django docs examples for `OneToOne` relationships all have this as `True`, so my gut is to go with it. It underscores the relationship between the two models.

I’ve only found one other place in platform where we do this: https://github.com/edx/edx-platform/blob/27ee8aee4af3f99be372a66d1a9c51615f485209/openedx/core/djangoapps/content/learning_sequences/models.py#L78-L80
—most of our `OneToOne` relationships (there are a handful) **don’t** have this set.

I’m happy to go with our default behavior, but I’d be even happier to better understand the reasoning in either direction.

Reply:
I'm not a domain expert here, but it seems like a reasonable thing to do, to me.

Reply:
Decided to stick with the default: will not add `primary_key=True`.